### ISSUE TITLE
Warning icon colour fails contrast checker

### ISSUE THREAD

**Issue by lyubomir-popov** [canonical-web-and-design/vanilla-framework#3587]
[ reported by @elioqoshi ]

#### White exclamation mark on yellow is not accessible:

![image](https://user-images.githubusercontent.com/2741678/109164588-8669f480-7772-11eb-9a10-c5c1fc5c2733.png)

I wonder if we can nudge the yellow to an accessible value, rather than doing the more straightforward solution of just flipping the text to black. Given the success and failure icons use white text, it would be odd to have dark on the yellow one only.
 
![image](https://user-images.githubusercontent.com/2741678/109164673-9bdf1e80-7772-11eb-93b2-ed9ea2a48262.png)

#### Minimal shift to make it accessible for graphical elements and large text:
![image](https://user-images.githubusercontent.com/2741678/109165756-d4332c80-7773-11eb-903d-1e49f7235f04.png)

![image](https://user-images.githubusercontent.com/2741678/109165894-fe84ea00-7773-11eb-806c-a5a01a819153.png)

#### 

For comparison, github warning:
![image](https://user-images.githubusercontent.com/2741678/109163538-59691200-7771-11eb-98e6-75c77ec5920b.png)

**Comment by spencerbygraves** [comment]
@lyubomir-popov perhaps we could think about using a dark "!" instead of changing the orange?

![image](https://user-images.githubusercontent.com/9105999/109281693-87ebf900-7814-11eb-95c9-8830bba1d4c6.png)

**Comment by lyubomir-popov** [comment]
Yes, but then the set "success / error / warning" will ahve white on red, white on green, then black on yellow - which is hardly ideal. Given this colour is not used for anything else and is not a brand colour, why not adjust it so it is fit for purpouse and doesn't force us to have inconsistencies within the semantic set?

**Comment by spencerbygraves** [comment]
@lyubomir-popov consistency of white on a colour is a fair point, but contrast might be more important.

Isn't the orange used on the icon our `color-caution`, which is also used on the label? So it would need to change there.

I'm not set on the idea, just wanted to see if we've considered it.

**Comment by lyubomir-popov** [comment]
Hmm, good point, if we have to put tex on top of it we'd have to go even darker... 
I guess this requires more thought. One other possible option to explore would be to make the red and green semantic colours lighter so they can support black...

**Comment by elioqoshi** [comment]
I understand both of these views, I don't think there is a right or wrong solution. Using a dark icon on light background might be less painful to apply in various situations, though it might not be the only solution.FOr reference, here is how Adobe's Spectrum Design System does it:
https://spectrum.adobe.com/page/alert-banner/

They specifically state to not use orange or yellow for soft errors as they are not accessible:
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/5436686/109290432-e373b200-7827-11eb-8651-dc44db62e70c.png">

A pretty bold move. They suggest using the neutral gray alerts instead. I suppose we don't want to do that, although it might solve this issue easily, especially if neutral notifications and soft errors have a similar semantic.

If we don't want to do that, we could also explore how GitHub does it of using a light signal colour as the background and consistently dark versions of these colours.
<img width="735" alt="image" src="https://user-images.githubusercontent.com/5436686/109291236-0357a580-7829-11eb-8778-f87c65b53a2b.png">
<img width="741" alt="image" src="https://user-images.githubusercontent.com/5436686/109291265-0e123a80-7829-11eb-984f-4baf0fb706ad.png">
<img width="734" alt="image" src="https://user-images.githubusercontent.com/5436686/109291306-1b2f2980-7829-11eb-9f87-0eda9c3bb31d.png">
https://styleguide.github.com/primer/components/alerts/

GitLab does a similar treatment to what @lyubomir-popov was planning:
https://design.gitlab.com/components/alert

<img width="912" alt="image" src="https://user-images.githubusercontent.com/5436686/109291340-26825500-7829-11eb-9feb-5ce74742fbb3.png">

For what it's worth, I think the GitLab or GitHub approach might be easiest and most consistent to maintain. A very light coloured background with dark or a very dark signal colour for text content and icons with optionally a saturated coloured line across the alert height

**Comment by lyubomir-popov** [comment]
Thanks @elioqoshi, nice benchmarking. I like the use of outlined icons (which we also have), they avoid the problem of contrast - as symbol and icon both use stroke. So if we have an accessible (near-brown) solid yellow, and a washed-out pale version of all 3 colours, then that seems to solve all problems? It also seems most pleasing to the eye to me.

**Comment by elioqoshi** [comment]
@lyubomir-popov Exactly my thoughts. That solution would tick all the boxes. I do think it looks best as well as the washed out signal colours for the background make it clear what kind of alert this is.

A saturated line for the left side of the alerts should probably be kept like in your proposal if they are mixed and stacked notifications. If they are not mixed it should be an optional thing I'd say.

**Comment by clagom** [comment]
Notification work to consider, modular components: https://app.zeplin.io/project/5fac06865cfff51271b72cc6

**Comment by clagom** [comment]
@SirSamTumless this is ready for planning

**Comment by lyubomir-popov** [comment]
Agreed to look at this during our visual design sprint.

**Comment by clagom** [comment]
@waynecrosby any news on this?

**Comment by waynecrosby** [comment]
@clagom Apologies it's dropped off our radar, I'll make sure it goes into the next iteration. Thanks

**Comment by clagom** [comment]
Thanks @waynecrosby, @cristinadresch @SirSamTumless can we please add it for planning thanks

**Comment by lyubomir-popov** [comment]
@waynecrosby @spencerbygraves suggested we work closely on this, let me know when you get to it and any way I can help.

**Comment by waynecrosby** [comment]
@lyubomir-popov @spencerbygraves I made a start and created a project in Figma, I've done a few different options based on the conversations above, I think the outlined versions are the most elegant solution. Take a look and let me know what you think:
https://www.figma.com/file/PDbgKIo0wd7JiwDw5diJW1/Accessible-Icon-colours?node-id=0%3A1

**Comment by cristinadresch** [comment]
@waynecrosby spencer is now off for 2 weeks, do you think we should put this one on hold for now? lyubo is also off this week, but back next

**Comment by waynecrosby** [comment]
@cristinadresch I am also on annual leave for 2 weeks so lets pick it up when we get back

**Comment by clagom** [comment]
@waynecrosby any update on this?

**Comment by waynecrosby** [comment]
@clagom We were going to wait for Spencer and myself to get back from annual leave, but I'll pick this up next week with Lyubo to progress it

**Comment by spencerbygraves** [comment]
@waynecrosby @lyubomir-popov I can see from the Figma doc that these options pass for Graphical and UI elements, but only one passes for text. Are we sure that this colour change doesn't affect other things that use text, like labels for instance? Thanks.
